### ISSUE TITLE
about the tokenize without input in prompt

### ISSUE THREAD

**Issue by moseshu** [tloen/alpaca-lora#425]
```
def tokenize(prompt, add_eos_token=True):
        # there's probably a way to do this with the tokenizer settings
        # but again, gotta move fast
        result = tokenizer(
            prompt,
            truncation=True,
            max_length=cutoff_len,
            padding=False,
            return_tensors=None,
        )
        if (
            result["input_ids"][-1] != tokenizer.eos_token_id
            and len(result["input_ids"]) < cutoff_len
            and add_eos_token
        ):
            result["input_ids"].append(tokenizer.eos_token_id)
            result["attention_mask"].append(1)

        result["labels"] = result["input_ids"].copy()

        return result
```
I was confused about above function. It's is different from  the below. the prompt design is promt_no_inpnut. 
the above code 'labels' is same to 'input_ids' .But in the below, the label  is padded  with value -100. 

the below code is from  https://github.com/tatsu-lab/stanford_alpaca/blob/main/train.py
```
IGNORE_INDEX=-100
def preprocess(
    sources: Sequence[str],
    targets: Sequence[str],
    tokenizer: transformers.PreTrainedTokenizer,
) -> Dict:
    """Preprocess the data by tokenizing."""
    examples = [s + t for s, t in zip(sources, targets)]
    examples_tokenized, sources_tokenized = [_tokenize_fn(strings, tokenizer) for strings in (examples, sources)]
    input_ids = examples_tokenized["input_ids"]
    labels = copy.deepcopy(input_ids)
    for label, source_len in zip(labels, sources_tokenized["input_ids_lens"]):
        label[:source_len] = IGNORE_INDEX
    return dict(input_ids=input_ids, labels=labels)
```

**Comment by OAfzal** [comment]
This is handled in the parent function, which is `generate_and_tokenize_prompt`

```
    def generate_and_tokenize_prompt(data_point):
        full_prompt = prompter.generate_prompt(
            data_point["instruction"],
            data_point["input"],
            data_point["output"],
        )
        tokenized_full_prompt = tokenize(full_prompt)
        if not train_on_inputs:
              ### <Look at this block. This is where they handle that case>

        return tokenized_full_prompt

```

**Comment by xpang-sf** [comment]
But train_on_inputs=True by default.
